src/ng/q.js

```diff
@@ -381,7 +381,11 @@ function qFactory(nextTick, exceptionHandler, errorOnUnhandledRejections) {
       if (!toCheck.pur) {
         toCheck.pur = true;
         var errorMessage = 'Possibly unhandled rejection: ' + toDebugString(toCheck.value);
-        exceptionHandler(errorMessage);
+        if (toCheck.value instanceof Error) {
+          exceptionHandler(toCheck.value, errorMessage);
```

Comment:
That sounds sensible but I'd still swap the order of operations. Note that currently we are trying to show the `'Possibly unhandled rejection'` message first and the error afterwards; we just put it all in one string argument so the stack trace gets lost. So I'd like:
```js
exceptionHandler('Possibly unhandled rejection', toCheck.value);
```
but that would require updating the docs to no longer require the cause to be the second parameter (no code changes required as it already works as I mentioned before).

I'd just want to know if there's any reason for the current signature so let's see what others say. :)